core/dnsserver/server_test.go

```diff
@@ -93,3 +111,149 @@ func BenchmarkCoreServeDNS(b *testing.B) {
 		s.ServeDNS(ctx, w, m)
 	}
 }
+
+func TestServer_WrapListener(t *testing.T) {
+	type fields struct {
+		Addr         string
+		server       [2]*dns.Server
+		m            sync.Mutex
+		zones        map[string]*Config
+		dnsWg        sync.WaitGroup
+		graceTimeout time.Duration
+		trace        trace.Trace
+		debug        bool
+		classChaos   bool
+	}
+	type args struct {
+		ln net.Listener
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		want   net.Listener
+	}{
+		{
+			name: "nil case",
+			fields: fields{
+				Addr:         ":53",
+				server:       [2]*dns.Server{},
+				m:            sync.Mutex{},
+				zones:        map[string]*Config{},
+				dnsWg:        sync.WaitGroup{},
+				graceTimeout: time.Second,
+				trace:        nil,
+				debug:        false,
+				classChaos:   false,
+			},
+			args: args{
+				ln: nil,
+			},
+			want: nil,
+		},
+		{
+			name: "non-nil case",
+			fields: fields{
+				Addr:         ":53",
+				server:       [2]*dns.Server{},
+				m:            sync.Mutex{},
+				zones:        map[string]*Config{},
+				dnsWg:        sync.WaitGroup{},
+				graceTimeout: time.Second,
+				trace:        nil,
+				debug:        false,
+				classChaos:   false,
+			},
+			args: args{
+				ln: &testHandler{},
+			},
+			want: &testHandler{},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := &Server{
+				Addr:         tt.fields.Addr,
+				server:       tt.fields.server,
+				m:            tt.fields.m,
+				zones:        tt.fields.zones,
+				dnsWg:        tt.fields.dnsWg,
+				graceTimeout: tt.fields.graceTimeout,
+				trace:        tt.fields.trace,
+				debug:        tt.fields.debug,
+				classChaos:   tt.fields.classChaos,
+			}
+			if got := s.WrapListener(tt.args.ln); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("Server.WrapListener() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestServer_Address(t *testing.T) {
```

Comment:
`Server.Address` is a single line function that returns simply Addr field.
It's too simple to warrant a test of this bulk.